src/pages/ActivationPage/ActivationPage.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import gql from 'graphql-tag'
+import { withApollo } from 'react-apollo'
+import { graphql, compose } from 'react-apollo'
+import { withRouter } from 'react-router-dom'
+import Section from '../../components/Section'
+import Container from '../../components/Container'
+import Alert from '../../components/Alert'
+import { Link } from 'react-router-dom'
+
+class ActivationPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      username: '',
+      activationCode: '',
+      errors: [],
+      successes: []
+    }
+  }
+  handleActivation = async e => {
+    const { username, activationCode } = this.state
+    try {
+      const { errors } = await this.props.activationMutation({
+        variables: {
+          username,
+          activationCode
+        }
+      })
+
+      if (errors && errors.length > 0) {
+        this.setState({ errors })
+      } else {
+        this.setState({ successes: [{ message: '邮件激活成功,请登录' }] })
+        //setTimeout(()=>{this.props.history.push('/login')},2000)
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+  componentDidMount() {
+    const searchParams = new URLSearchParams(this.props.location.search)
+    this.state.username = searchParams.get('username')
+    this.state.activationCode = searchParams.get('activationcode')
+    this.handleActivation()
+  }
+  render() {
+    return (
+      <Container small>
+        <Section padded>
+          <h2>激活邮箱</h2>
+          {this.state.errors.map((error, index) => {
+            let template = <Alert key={index} message={error.message} error />
```

Comment:
Initializing the template at the very beginning is sometimes redundant, we should return the template based on the error code:

```js
if (error.code == '223') {
  return (
    <Alert key={index} message={error.message} error>
      <br />
      <Link to="/sendactivation">立即发送</Link>
    </Alert>
  )
} else {
  return <Alert key={index} message={error.message} error />
}
```